### PR TITLE
[Core] Add iga_test_factory

### PR DESCRIPTION
**Pull request by tteschemacher**
Hi Iga people,

I am adding here a test for an IGA example. This is the next milestone to be able to simulate this.

### CODE REVIEW

**Comment by tteschemacher** [comment]
@dagmawibekel could you have a look at this? We need to merge the CadIoModeler first but then we can merge this one. Do you agree with this concept? It is adapted from the StructuralMechanicsApplication.

**Comment by tteschemacher** [comment]
@philbucher is this text_factory which is obviously adapted from the structural mechanics application the best way to go, or would you suggest me any modification, here?

Other than that, I would be fine to merge it as soon as possible.

File: applications/IgaApplication/tests/single_patch_test/single_patch_project_parameters.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        "move_mesh_flag": true,
+        "convergence_criterion": "residual_criterion",
+        "displacement_relative_tolerance": 0.0001,
+        "displacement_absolute_tolerance": 1E-09,
+        "residual_relative_tolerance": 0.0001,
+        "residual_absolute_tolerance": 1E-09,
+        "max_iteration": 1,
+        "solver_type": "static",
+        "linear_solver_settings": {
+            "solver_type": "amgcl",
```
Comment (philbucher) [inline applications/IgaApplication/tests/single_patch_test/single_patch_project_parameters.json:35]: do you need amgcl here?

**Comment by philbucher** [comment]
> @philbucher is this text_factory which is obviously adapted from the structural mechanics application the best way to go, or would you suggest me any modification, here?

well "adapted" is a bit of a stretch don't you think... It is a plain old copy-paste :laughing: 

TBH I don't love it but I haven't had time to refactor it yet in StructuralMechanics. I would give it a go as is, maybe we can refactor it later...

approval I leave to @dagmawibekel

File: applications/IgaApplication/tests/single_patch_test/single_patch_project_parameters.json (hunk 1 above)
Comment (tteschemacher) [reply applications/IgaApplication/tests/single_patch_test/single_patch_project_parameters.json:35]: no, which one would be recommended?

Comment (philbucher) [reply applications/IgaApplication/tests/single_patch_test/single_patch_project_parameters.json:35]: I would not specify it at all and use the solution from the StructuralMechanics Test-Factory

**Comment by tteschemacher** [comment]
@dagmawibekel can we give it a lift?

**Comment by tteschemacher** [comment]
ping @dagmawibekel . can we merge it?

File: applications/IgaApplication/tests/iga_test_factory.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
```
Comment (dagmawibekel) [inline applications/IgaApplication/tests/iga_test_factory.py]: do we still need this? we can also delete it later!

**Review by dagmawibekel — DISMISSED** [review]

File: applications/IgaApplication/tests/iga_test_factory.py (hunk 2 above)
Comment (tteschemacher) [reply applications/IgaApplication/tests/iga_test_factory.py]: I actually don't know. It used to be standard. Isn't it needed anymore @philbucher ?

**Comment by tteschemacher** [comment]
thanks 👍

File: applications/IgaApplication/tests/iga_test_factory.py (hunk 2 above)
Comment (philbucher) [reply applications/IgaApplication/tests/iga_test_factory.py]: we do no longer need this, Python2 support is dropped :D

**Review by dagmawibekel — APPROVED** [review]

File: applications/IgaApplication/tests/iga_test_factory.py (hunk 2 above)
Comment (tteschemacher) [reply applications/IgaApplication/tests/iga_test_factory.py]: alright. Thanks. I removed it.

**Pull request merged**